File: python/heap-priority-queue/kth-largest-element-in-an-array.py
import queue
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        q = queue.PriorityQueue()

        for i in nums:
            q.put(-i)

        for i in range(k):
            current = -q.get()

        return current

File: c++/heap-priority-queue/kth-largest-element-in-an-array.cc
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        std::priority_queue<int> pq;
        for (int num : nums) {
            pq.push(num);
        }

        for (int i = 0; i < k - 1; ++i) {
            pq.pop();
        }

        return pq.top();
    }
};
